### PR TITLE
Refactor: update eslint and it's packages

### PR DESCRIPTION
**Pull request by elukoyanov**
## Motivation/Description of the PR
- Description of this PR, which problem it solves
- A link to the corresponding issue (if applicable).

Updated packages
- eslint
- eslint-config-airbnb-base
- eslint-plugin-import
- eslint-plugin-mocha

## Type of change

- [x] Dependencies changes/updates
- [ ] Breaking changes
- [ ] New functionality
- [ ] Bug fix
- [x] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [ ] Tests have been added
- [x] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "import/no-unresolved": 0,
     "no-await-in-loop": 0,
     "arrow-body-style": 0,
-    "no-loop-func": 0
+    "no-loop-func": 0,
+    "arrow-parens": 0
```
Comment (elukoyanov) [inline .eslintrc.json]: Had to disable rule, as almost all usages of functions with one argument are without parens

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 25
           

Clones added
============
- lib/command/init.js  2
- lib/command/generate.js  2
- lib/plugin/puppeteerCoverage.js  1
- lib/command/configMigrate.js  1
- lib/plugin/stepByStepReport.js  1
- lib/plugin/screenshotOnFail.js  1
         

Clones removed
==============
+ lib/helper/SeleniumWebdriver.js  -1
+ lib/command/gherkin/steps.js  -1
+ test/rest/REST_test.js  -1
+ test/rest/ApiDataFactory_test.js  -1
+ lib/command/gherkin/snippets.js  -1
+ lib/helper/Nightmare.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4754942&bid=15859067)

File: lib/helper/WebDriver.js [hunk 2]
```diff
@@ -2186,58 +2193,57 @@ async function proceedSee(assertType, text, context, strict = false) {
...
-// @param {function} callback Async function to excute on each array item
-// @return Array
-async function forEachAsync(array, callback, option = {}) {
+/**
+ * Mimic Array.forEach() API, but with an async callback function.
+ * Execute each callback on each array item serially. Useful when using WebDriver API.
+ *
+ * Added due because of problem with chrome driver when too many requests
+ * are made simultaneously. https://bugs.chromium.org/p/chromedriver/issues/detail?id=2152#c9
+ * @private
```
Comment (elukoyanov) [inline lib/helper/WebDriver.js]: Made JSDoc format.

But it leads to creation definitions for this internal functions.
Added such code ignore by `@private` and `@ignore` tags in #2113

File: .eslintrc.json (hunk 1 above)
Comment (DavertMik) [reply .eslintrc.json]: I agree, that looks nicer

**Comment by DavertMik** [comment]
So far I like it, however, merging it will make conflict for next PRs.

**Comment by elukoyanov** [comment]
Which one?
Write after merges, so I'll update this

File: lib/helper/WebDriverIO.js [hunk 3]
```diff
@@ -1877,16 +1881,11 @@ async function forEachAsync(array, callback, option = {}) {
...
-    let res;
-    try {
-      res = await callback(inputArray[index], index, inputArray);
-      if (Array.isArray(res) && expandArrayResults) {
-        res.forEach(val => values.push(val));
-      } else if (res) {
-        values.push(res);
-      }
-    } catch (err) {
-      throw err;
```
Comment (PeterNgTr) [inline lib/helper/WebDriverIO.js]: don't we need the error catching anymore?

Comment (PeterNgTr) [reply lib/helper/WebDriverIO.js]: oh nvm, we will drop this helper at some point.

File: package.json [hunk 4]
```diff
@@ -86,10 +86,10 @@
...
     "chai-subset": "^1.6.0",
     "documentation": "^8.1.2",
-    "eslint": "^4.17.0",
-    "eslint-config-airbnb-base": "^12.1.0",
-    "eslint-plugin-import": "^2.18.2",
-    "eslint-plugin-mocha": "^5.3.0",
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb-base": "^14.0.0",
+    "eslint-plugin-import": "^2.19.1",
+    "eslint-plugin-mocha": "^6.2.2",
```
Comment (PeterNgTr) [inline package.json]: may we consider using https://greenkeeper.io/ to keep us more peace?

File: lib/helper/WebDriverIO.js (hunk 3 above)
Comment (elukoyanov) [reply lib/helper/WebDriverIO.js]: we catch `catch (err)` and then throw again `throw err` without any change. So this catch is useless

File: package.json (hunk 4 above)
Comment (elukoyanov) [reply package.json]: don't know.
create an feature issue

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 62
           

Clones added
============
- lib/command/init.js  2
- lib/plugin/screenshotOnFail.js  1
- lib/command/configMigrate.js  1
- lib/plugin/stepByStepReport.js  1
- lib/plugin/puppeteerCoverage.js  1
- lib/command/generate.js  2
         

Clones removed
==============
+ lib/command/gherkin/steps.js  -1
+ lib/helper/SeleniumWebdriver.js  -1
+ test/rest/REST_test.js  -1
+ lib/command/gherkin/snippets.js  -1
+ test/rest/ApiDataFactory_test.js  -1
+ lib/helper/Nightmare.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4754942&bid=15859067)

**Comment by elukoyanov** [comment]
pulled master, please, re-review @DavertMik , @PeterNgTr

**Comment by DavertMik** [comment]
Thank you!

**Pull request merged**